src/errors.rs

```diff
@@ -18,7 +41,17 @@ impl std::fmt::Display for ParseError {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
             ParseError::InsufficientData => write!(f, "Insufficient data was provided"),
-            ParseError::ParseError => write!(f, "Error while parsing"),
+            ParseError::ConversionFailed => {
+                write!(f, "A value conversion failed due to out of range data")
+            }
+            ParseError::InvalidExpression => write!8
```

Comment:
Here's the culprit :)
Error messages from rustfmt are completely useless sometimes.

Reply:
Thanks. I totally missed that one. :facepalm: